server/mqtt.go

```diff
@@ -4550,6 +4616,31 @@ func mqttIsRetained(flags byte) bool {
 	return flags&mqttPubFlagRetain != 0
 }
 
+// Checks if the subject is in the form of a Sparkplug BIRTH or DEATH message,
+// MQTT topic like spBv1.0/{group_id}/{message_type}/{edge_node_id}/{device_id}
+func sparkbIsBirth(natsSubject, prefix []byte) (is bool, typ, groupID, nodeID, deviceID string) {
+	if !bytes.HasPrefix(natsSubject, prefix) {
+		return false, "", "", "", ""
+	}
+	natsSubject = natsSubject[len(prefix):]
+	parts := bytes.Split(natsSubject, []byte("."))
+	if len(parts) < 3 {
+		return false, "", "", "", ""
+	}
+	typ = string(parts[1])
+	isDevice := (typ == sparkbDBIRTH)
+	if typ != sparkbNBIRTH && !isDevice {
+		return false, "", "", "", ""
+	}
+
+	groupID = string(natsSubjectToMQTTTopic(parts[0]))
```

Comment:
Why do we have to convert? Meaning do we expect messages to be from NATS? I guess it does not hurt because if they are already "MQTT topic-like" tokens, then should not be any conversion needed.

Reply:
I just refactored this in a separate branch where I worked on overwriting the DEATH timestamp. Once I implement all other feedback, I'll see if it makes sense to merge it here, or submit as a separate PR.

Reply:
See https://github.com/nats-io/nats-server/pull/5251; 1/5 I prefer to keep it a separate PR, it will probably have a number of comments of its own. But if it makes it easier for you to review - feel free to merge into this branch.